src/PortingEmbed/PortingFormContainer.tsx

```diff
@@ -52,9 +50,20 @@ export function PortingFormContainer({
   useEffect(() => {
     if (!error) return
 
-    onError?.(new Error(error))
+    if (error instanceof ApiError) {
+      onError(error, {
+        code: error instanceof ApiError ? error.code : 'initializationError',
```

Comment:
You're already in a clause where the error is of type `ApiError` 😉 
```suggestion
        code: error.code
```